script.js

```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
     })
     stringForHtml += '</p>'
     textToTypeContainer.innerHTML += stringForHtml
+    let upcomingWord = document.getElementById(0)
+    upcomingWord.style.color = ("#cc7a00")
+    upcomingWordLength = upcomingWord.clientWidth
+    totalLengthOfWords += upcomingWordLength
 })
 
 
 
+textInput.addEventListener('keyup', event => {
+    if (event.code === 'Space') {
+        let textValue = textInput.value
+        let textValueWithoutSpace = textValue.slice(0, textValue.length - 1);
+        wordIndexCount++
+        wordIsValid = (textValueWithoutSpace === referenceTextArray[wordIndexCount])
+
+        let wordJustFinished = document.getElementById(wordIndexCount)
+        if (wordIsValid) {
+            numberOfValidWords++ // for later use
+            textInput.value = ''
```

Comment:
this line and line 51 dont need to be in the if/else as it happens in both scenarios

Reply:
I moved line 48 outside if the if/else and removed line 51